### ISSUE TITLE
Press Esc Key on modal generate a jquery UI error

### ISSUE THREAD

**Issue by apomili** [magento/magento2#14593]
If you add a product in the cart, then you press the trash icon to delete it a modal window will appear.
If you press the "Esc" key, this js error is raised

- Precondition: I suppose every version of Magento with Luma Theme,
like this demo http://demo-acm2.bird.eu   (ver 2.2.3)

- Step to reproduce: Add a product in the cart. From the mini cart in the upper right click the trash icon. A confirmation modal will open. Press Esc button

-  Expected result: Modal close

- Actual result: Uncaught TypeError: Cannot read property 'stopImmediatePropagation' of undefined
    at Object.always (sidebar.js:81)

**Comment by magento-engcom-team** [comment]
Hi @apomili thank you for your report.
Please use the issue [reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) to report an issue. Please, edit issue description and include the preconditions, the steps required to reproduce, the actual result, and the expected result.
Please, add the following information:
 - [x] Preconditions: Magento version
 - [x] Steps to reproduce
 - [x] Expected result
 - [x] Actual result

**Comment by apomili** [comment]
Done

**Comment by 0m3r** [comment]
reproduced on Magento 2.2.4

 Preconditions: Magento version 2.2.4 with luma theme

Step to reproduce: 
 - Open chrome developer console 
 - Add a product to the cart.
 - From the mini cart in the upper right click the trash icon. 
 - A confirmation modal will open. 
 - Press Esc button

Expected Result
 - Close the confirm 

Actual result
  - js error 

https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Checkout/view/frontend/web/js/sidebar.js#L96

~~~
Uncaught TypeError: Cannot read property 'stopImmediatePropagation' of undefined
~~~
![2018-06-21 15-50-01](https://user-images.githubusercontent.com/412612/41720145-cbfd9e28-756a-11e8-9c4a-1500171c3788.png)

**Comment by 0m3r** [comment]
Proposal fix 

https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Checkout/view/frontend/web/js/sidebar.js#L96

~~~js 
                        always: function (e) {
                            if (e && typeof e.stopImmediatePropagation === 'function') {
                                e.stopImmediatePropagation();
                            }
                        }
~~~

**Comment by VladimirZaets** [comment]
Hi @apomili. Thank you for your report. 
The issue has been fixed in magento/magento2#16477 by @0m3r in 2.2-develop branch
Related commit(s):
 - [8399f63ab93c361349b4f681b3817e59ce35abc3](https://github.com/magento/magento2/commit/8399f63ab93c361349b4f681b3817e59ce35abc3)
 - [a6a17d317255e472725fcec5e585154041c8de68](https://github.com/magento/magento2/commit/a6a17d317255e472725fcec5e585154041c8de68)
 - [21f85b47263ebcea478457bc8827e2620f8d90be](https://github.com/magento/magento2/commit/21f85b47263ebcea478457bc8827e2620f8d90be)

The fix will be available with the upcoming 2.2.6 release.

**Comment by sidolov** [comment]
Hi @apomili. Thank you for your report. 
The issue has been fixed in magento/magento2#16623 by @mageprince in 2.3-develop branch
Related commit(s):
 - [0fdc0a87d96608380790f298343eb9ae7e0886ee](https://github.com/magento/magento2/commit/0fdc0a87d96608380790f298343eb9ae7e0886ee)
 - [07f7d83d3ec7658fe678a07e698d3ec2f6513cf4](https://github.com/magento/magento2/commit/07f7d83d3ec7658fe678a07e698d3ec2f6513cf4)
 - [160bc5d6ee5751ca0cdd71a828b11315270bf958](https://github.com/magento/magento2/commit/160bc5d6ee5751ca0cdd71a828b11315270bf958)

The fix will be available with the upcoming 2.3.0 release.

**Comment by sidolov** [comment]
Hi @apomili. Thank you for your report. 
The issue has been fixed in magento/magento2#17223 by @gelanivishal in 2.1-develop branch
Related commit(s):
 - [b50caeb31de1c71a44d4a52af980494126246575](https://github.com/magento/magento2/commit/b50caeb31de1c71a44d4a52af980494126246575)
 - [540cdb5e13097fbf3ffd5edd7a23beb8d88e12f2](https://github.com/magento/magento2/commit/540cdb5e13097fbf3ffd5edd7a23beb8d88e12f2)
 - [d937ca701cd217d959b43f009e192c69cd90a595](https://github.com/magento/magento2/commit/d937ca701cd217d959b43f009e192c69cd90a595)

The fix will be available with the upcoming 2.1.15 release.
